### PR TITLE
952 value card wrap units for small sizes

### LINKED ISSUE #952: [ValueCard] Wrap units instead of hiding at small screen sizes.
Currently the ValueCard will hide all tangential information other than the value, so units and thresholds,  at smaller sizes. Card needs to have these values wrap instead of hiding them. 

Must check all Value card stories to be sure content is not pushed off the card.

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #952

**Summary**
Instead of hiding units and thresholds icons at smaller sizes the ValueCard now allows them to wrap  for attributes with a measured width <= 100px. For vertical cards with wrapping enabled and more than 2 items listed the size of the value and the units are decreased to fit the available height.

**Change List (commits, features, bugs, etc)**
- Added SCSS files `_value-card.scss`, `_attribute.scss`, `_unit-renderer.scss` and `_value-renderer.scss` to house new CSS needed
- Refactored styled components for Unit-renderer info the _unit-renderer.scss
- Added a new story that shows wrapping and modified story  "tall / vertical / 6" to show compact wrapping.

_Note_
Attribute alignments in Attribute.jsx are achieved by different margins and we should probably use flex instead, but I didn't dare to refactor that code. Also, I think there is an over use of wrapping divs but I didn't dare to change that either due to fear of unknown downstream effects.

**Acceptance Test (how to verify the PR)**
Tests the different card stories with a wrapper width of 120 to trigger the value and threshold icon wrapping.

### CODE REVIEW

File: src/components/ValueCard/Attribute.jsx [hunk 1]
```diff
@@ -84,6 +78,7 @@ const propTypes = {
   ),
   renderIconByName: PropTypes.func,
   precision: PropTypes.number,
+  attributeCount: PropTypes.number.isRequired,
```
Comment (sstone2423) [inline src/components/ValueCard/Attribute.jsx]: Prop comments would be helpful

File: src/components/ValueCard/ValueCard.story.jsx [hunk 2]
```diff
@@ -72,6 +72,31 @@ storiesOf('Watson IoT|ValueCard', module)
       </div>
     );
   })
+
+  .add('small / wrapping', () => {
```
Comment (sstone2423) [inline src/components/ValueCard/ValueCard.story.jsx]: There is a ton of stories already. Is there any way to combine this with another story?

File: src/components/ValueCard/ValueRenderer.jsx [hunk 3]
```diff
@@ -17,6 +21,8 @@ const propTypes = {
   size: PropTypes.string.isRequired,
   color: PropTypes.string,
   isVertical: PropTypes.bool,
+  allowedToWrap: PropTypes.bool.isRequired,
```
Comment (sstone2423) [inline src/components/ValueCard/ValueRenderer.jsx]: Prop comments would be helpful

File: src/components/ValueCard/_unit-renderer.scss [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  padding-left: 0.25rem;
+  font-weight: lighter;
+  white-space: nowrap;
+  padding-bottom: 0.25rem;
+  font-size: var(--default-font-size);
+}
+
+.#{$iot-prefix}--value-card__attribute-unit--mini {
+  padding-bottom: 0;
+  font-size: 1rem;
```
Comment (sstone2423) [inline src/components/ValueCard/_unit-renderer.scss]: Needs carbon styling variables

File: src/components/ValueCard/_value-renderer.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
+@import '../../globals/vars';
+
+.#{$iot-prefix}--value-card__attribute-value--wrappable {
+  margin-right: 0.25rem;
```
Comment (sstone2423) [inline src/components/ValueCard/_value-renderer.scss]: Needs styling variables

File: src/components/ValueCard/_unit-renderer.scss (hunk 4 above)
Comment (bjornalm) [reply src/components/ValueCard/_unit-renderer.scss]: Done. 
Can I just assume that it is always safe translate 1rem into $carbon--spacing-05 or are there any special considerations needed given that the 1rem value of $carbon--spacing-05 is defined as a SCSS default and can be overridden?

File: src/components/ValueCard/ValueRenderer.jsx (hunk 3 above)
Comment (bjornalm) [reply src/components/ValueCard/ValueRenderer.jsx]: done

File: src/components/ValueCard/ValueCard.story.jsx (hunk 2 above)
Comment (bjornalm) [reply src/components/ValueCard/ValueCard.story.jsx]: Ok, merged it into the "small / units" story that now shows to cards

File: src/components/ValueCard/Attribute.jsx (hunk 1 above)
Comment (bjornalm) [reply src/components/ValueCard/Attribute.jsx]: Good point

File: src/components/ValueCard/_value-renderer.scss (hunk 5 above)
Comment (bjornalm) [reply src/components/ValueCard/_value-renderer.scss]: done

File: src/components/ValueCard/_unit-renderer.scss (hunk 4 above)
Comment (sstone2423) [reply src/components/ValueCard/_unit-renderer.scss]: As far as I've seen / know, its safe to stick to those variables. I've personally never seen them change, but @davidicus or @tay1orjones could maybe give a better answer

Comment (tay1orjones) [reply src/components/ValueCard/_unit-renderer.scss]: Yes, typically it's a safe bet to use `$carbon--spacing-05` in lieu of `1rem` on layout properties (`padding`, `margin`, `top`, etc). Though it's worth considering if the variable value should change, how that may impact or break the component. By and large we should be using the variables and avoid hardcoded values.

In this case though, font settings should generally be defined by `@include`ing the [typography variables](https://www.carbondesignsystem.com/guidelines/typography/productive), not spacing variables. In this case we might want to use `productive-heading-02`?

`@include type-style('productive-heading-02');`

**Comment by bjornalm** [comment]
> Looks good pending that comment about potentially needing typography styles via the Carbon variable.
> 
> This story in particular seemed to be a good demonstration of the difference at 120px wide:
> 
> https://deploy-preview-957--carbon-addons-iot-react.netlify.com/?path=/story/watson-iot-valuecard--medium-vertical-3
> 
> https://carbon-addons-iot-react.com/?path=/story/watson-iot-valuecard--medium-vertical-3

Yes indeed

**Pull request merged**